pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Stage.groovy

```diff
@@ -98,7 +110,7 @@ class Stage implements Serializable {
         this.input = input
         this.stages = stages
         this.parallel = parallel
-        this.matrix = matrix
```

Comment:
The matrix field on runtime was serving no purpose but making code more complex.  `Matrix` extends `Parallel` and at runtime they are equivalent.